gateway/middleware.go

```diff
@@ -692,17 +692,20 @@ func (t BaseMiddleware) CheckSessionAndIdentityForValidKey(originalKey *string,
 		return session.Clone(), true
 	}
 
-	t.Logger().Debug("Querying authstore")
-	// 2. If not there, get it from the AuthorizationHandler
-	session, found = t.Spec.AuthManager.KeyAuthorised(key)
-	if !found && storage.TokenOrg(key) != t.Spec.OrgID {
-		//treat it as a custom key
-		key = generateToken(t.Spec.OrgID, key)
-		cacheKey = key
-		if t.Spec.GlobalConfig.HashKeys {
-			cacheKey = storage.HashStr(cacheKey)
-		}
+	// Only search in RPC if the storage is RPCStorageHandler and it's not in emergency mode
+	if _, ok := t.Spec.AuthManager.Store().(*RPCStorageHandler); ok && !rpc.IsEmergencyMode() {
```

Comment:
Now it will not run if Store is a different type :) 

You probably need smth like: !ok || !rpc.IsEmergencyMode()